python/lsst/ip/diffim/imageDecorrelation.py

```diff
@@ -52,188 +54,211 @@ def computeClippedImageStats(im):
-    @param exposure[in] the science afwImage.Exposure used for PSF matching
-    @param subtractedExposure[in] the subtracted exposure produced by 
-    ip_diffim.ImagePsfMatchTask.subtractExposures()
-    @param psfMatchingKernel an (optionally spatially-varying) PSF matching kernel produced 
-    by ip_diffim.ImagePsfMatchTask.subtractExposures()
-    @param log an input pexLog for outputting messages
-
-    @return the the decorrelated diffim and decorrelation correction kernel (which may be ignored)
-    The returned decorrelated diffim has an updated PSF as well.
-
-    @note the subtractedExposure is NOT updated
-    @note Here we currently convert a spatially-varying matching kernel into a constant kernel, 
-    just by computing it at the center of the image (tickets DM-6243, DM-6244).
-    @note We are also using a constant accross-the-image measure of sigma (sqrt(variance)) to compute
-    the decorrelation kernel.
-    @note Still TBD (ticket DM-6580): understand whether the convolution is correctly modifying 
-    the variance plane of the new subtractedExposure.
-    """
-    kimg = None
-    try:
-        if psfMatchingKernel.isSpatiallyVarying():
-            spatialKernel = psfMatchingKernel
-            kimg = afwImage.ImageD(spatialKernel.getDimensions())
-            bbox = subtractedExposure.getBBox()
-            xcen = (bbox.getBeginX() + bbox.getEndX()) / 2.
-            ycen = (bbox.getBeginY() + bbox.getEndY()) / 2.
-            spatialKernel.computeImage(kimg, True, xcen, ycen)
-    except:  # Not a spatially-varying kernel (usually not true)
-        kimg = psfMatchingKernel.computeImage()
-
-    # Compute the images' sigmas (sqrt of variance)
-    _, _, sig1squared = computeClippedImageStats(templateExposure.getMaskedImage())
-    sig1 = np.sqrt(sig1squared)
-
-    _, _, sig2squared = computeClippedImageStats(exposure.getMaskedImage())
-    sig2 = np.sqrt(sig2squared)
-
-    corrKernel = computeDecorrelationKernel(kimg.getArray(), sig1=sig1, sig2=sig2)
-    fcorrKernel = fixEvenKernel(corrKernel)
-    log.info("Decorrelation: Convolving.")
-    correctedExposure, corrKern = doConvolve(subtractedExposure, fcorrKernel)
-    log.info("Decorrelation: Finished with convolution.")
-
-    # Compute the subtracted exposure's updated psf
-    psf = subtractedExposure.getPsf().computeImage().getArray()
-    psfc = computeCorrectedDiffimPsf(fcorrKernel, psf, sig1=sig1, sig2=sig2)
-    psfcI = afwImage.ImageD(psfc.shape[0], psfc.shape[1])
-    psfcI.getArray()[:, :] = psfc
-    psfcK = afwMath.FixedKernel(psfcI)
-    psfNew = measAlg.KernelPsf(psfcK)
-    correctedExposure.setPsf(psfNew)
-
-    return correctedExposure, corrKern
+            kernelImg = afwImage.ImageD(fkernel.shape[0], fkernel.shape[1])
+            kernelImg.getArray()[:, :] = fkernel
+            kern = afwMath.FixedKernel(kernelImg)
+            maxloc = np.unravel_index(np.argmax(fkernel), fkernel.shape)
```

Comment:
I don't understand what the safe thing to do is. If the kernel was reliably computed with the center rounded up (since LSST rounds it down, and it's off by one) then you really ought to just set the center to the correct value (and trim if you feel you must, but that is strictly optional). If the high point can be anywhere then this makes sense. What worries me is if the center is always up and to the right, but the computed kernel is weird for some reason, then your measurement of the peak may do the wrong thing.